packages/edit-navigation/src/store/test/resolvers.js

```diff
@@ -28,288 +37,163 @@ jest.mock( '@wordpress/blocks', () => {
 	};
 } );
 
-describe( 'getNavigationPostForMenu', () => {
-	it( 'returns early when a menuId is not provided', () => {
-		const generator = getNavigationPostForMenu( null );
-		expect( generator.next().value ).toBeUndefined();
-		expect( generator.next().done ).toBe( true );
-	} );
-
-	it( 'gets navigation post for menu id', () => {
-		const menuId = 123;
+function createRegistryWithStores() {
+	// create a registry and register stores
+	const registry = createRegistry();
+
+	registry.register( editNavigationStore );
+	registry.register( coreStore );
+
+	// Set up the navigation post entity.
+	registry.dispatch( coreStore ).addEntities( [
+		{
+			kind: NAVIGATION_POST_KIND,
+			name: NAVIGATION_POST_POST_TYPE,
+			transientEdits: { blocks: true, selection: true },
+			label: 'Navigation Post',
+			__experimentalNoFetch: true,
+		},
+	] );
 
-		const generator = getNavigationPostForMenu( menuId );
+	return registry;
+}
 
-		const id = buildNavigationPostId( menuId );
-		const stubPost = {
-			id,
-			slug: id,
-			status: 'draft',
-			type: 'page',
-			blocks: [],
-			meta: {
-				menuId,
-			},
-		};
+const mockMenuItems = [
+	{
+		id: 100,
+		title: {
+			raw: 'wp.com',
+			rendered: 'wp.com',
+		},
+		url: 'http://wp.com',
+		menu_order: 1,
+		menus: 1,
+		parent: 0,
+		object_id: 123,
+		object: 'post',
+		classes: [ 'menu', 'classes' ],
+		xfn: [ 'nofollow' ],
+		description: 'description',
+		attr_title: 'link title',
+	},
+	{
+		id: 101,
+		title: {
+			raw: 'wp.org',
+			rendered: 'wp.org',
+		},
+		url: 'http://wp.org',
+		menu_order: 2,
+		menus: 1,
+		parent: 0,
+		object_id: 456,
+		object: 'page',
+		classes: [],
+		xfn: [],
+		description: '',
+		attr_title: '',
+		target: '_blank',
+	},
+	{
+		id: 102,
+		title: {
+			raw: 'My Example Page',
+			rendered: 'My Example Page',
+		},
+		url: '/my-example-page/',
+		object_id: 56789,
+		type: 'post-type',
+		menu_order: 3,
+		menus: 1,
+		parent: 0,
+		classes: [],
+		object: 'custom',
+		xfn: [],
+		description: '',
+		attr_title: '',
+		target: '_blank',
+	},
+];
 
-		// Create stub post.
-		expect( generator.next().value ).toEqual(
-			dispatch(
-				'core',
-				'receiveEntityRecords',
-				NAVIGATION_POST_KIND,
-				NAVIGATION_POST_POST_TYPE,
-				stubPost,
-				{ id: stubPost.id },
-				false
-			)
-		);
+describe( 'getNavigationPostForMenu', () => {
+	jest.useRealTimers();
```

Comment:
Aren't real timers on by default?

Reply:
That's what the documentation says, but the tests fail without this line. Do you think it's worth diving into?

Reply:
How bizarre! No, not a blocker for this PR. Maybe @gziolo knows what is going on?